drivers/serial: stm32: Simplify code dependency checks

erwango opened the pull request:
Rework stm32 serial driver to simplify code activation depending on series.
Since more and more series are present in Zephyr, prefer use of definitions from STM32Cube packages rather than series Kconfig symbols which will require long and heavy `#ifdef` lines

erwango committed:
drivers/serial: stm32: simplify check of TEACK/REACK flags

Remove reference to SoC series in activation of TEACK/REACK flags
checks. Use flags definitions instead which is defined, if supported,
in STM32Cube packages.
Decouple the checks since REACK is not supported in some series.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango commented:
@zhaozhongchen , @pic16f887, FYI

ydamigos reviewed (approved):
LGTM

lpereira commented on drivers/serial/uart_stm32.c:
```diff
@@ -331,12 +331,17 @@ static int uart_stm32_init(struct device *dev)
+#endif /* !USART_ISR_TEACK */
+
+#ifdef USART_ISR_REACK
+	/* Wait until TEACK flag is set */
```
This is just the comment, but shouldn't this be the REACK flag?

erwango replied:
Indeed, thanks!

erwango committed:
drivers/serial: stm32: Put LPUART code under LPUART Kconfig symbol

In order to ease integration of new series, remove reference to
series in code activation. Use LPUART support Kconfig symbol instead.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>
